### PR TITLE
fission: init at 1.12.0

### PR DESCRIPTION
**Pull request by NeverBehave**
###### Motivation for this change

Add the cli used by end user to interact [Fission](https://fission.io).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/fission/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  postInstall = ''
+    mv $out/bin/fission-cli $out/bin/fission
+  '';
+
+  meta = with lib; {
+    description = "The cli used by end user to interact Fission";
+    homepage = "https://fission.io";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ neverbehave ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fission/default.nix:29]:
```suggestion
```

set by buildGoModule

File: pkgs/development/tools/fission/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  vendorSha256 = "12clw0wy4lypf45imqnabj39yxqpi348csr4m5d0d1rksxgvwngq";
+
+  buildFlagsArray = "-ldflags=-X info.Version=${version}";
+
+  subPackages = [ "cmd/fission-cli" ];
+
+  postInstall = ''
+    mv $out/bin/fission-cli $out/bin/fission
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fission/default.nix:21]:
```suggestion
    ln -s $out/bin/fission-cli $out/bin/fission
```

File: pkgs/development/tools/fission/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "fission";
+    repo = "fission";
+    rev = version;
+    sha256 = "0izvkjd7ydcxhr6zmgrbfm3ybz2kf4p27099lr07gd4x7c6xxmqr";
+  };
+
+  vendorSha256 = "12clw0wy4lypf45imqnabj39yxqpi348csr4m5d0d1rksxgvwngq";
+
+  buildFlagsArray = "-ldflags=-X info.Version=${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fission/default.nix:16]:
```suggestion
  buildFlagsArray = "-ldflags=-s -w -X info.Version=${version}";
```

**Comment by NeverBehave** [comment]
ping @SuperSandro2000

File: pkgs/development/tools/fission/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ lib, fetchFromGitHub, buildGoModule, ... }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fission/default.nix:1]:
```suggestion
{ lib, fetchFromGitHub, buildGoModule }:
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116148` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fission</li>
</ul>
</details>

**Pull request merged**